val.py

```diff
@@ -302,10 +312,11 @@ def run(
             if single_cls:
                 pred[:, 5] = 0
             predn = pred.clone()
+            pred_clone = pred.clone()
             scale_boxes(im[si].shape[1:], predn[:, :4], shape, shapes[si][1])  # native-space pred
 
             # Evaluate
-            if nl:
+            if nl and not skip_evaluation:
```

Comment:
I think you need to have not skip_evaluation above the orig if statement because of line 329